activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -519,9 +519,16 @@ def in_order_of(column, values)
       values = values.map { |value| type_caster.type_cast_for_database(column, value) }
       arel_column = column.is_a?(Symbol) ? order_column(column.to_s) : column
 
+      where_clause =
+        if values.include?(nil)
+          arel_column.in(values.compact).or(arel_column.eq(nil))
+        else
+          arel_column.in(values)
+        end
+
       spawn
-        .order!(connection.field_ordered_value(arel_column, values))
-        .where!(arel_column.in(values))
+        .order!(field_ordered_value(arel_column, values))
+        .where(where_clause)
```

Comment:
Was there a reason to drop the `!` ?

```suggestion
        .where!(where_clause)
```